server/sonar-server/src/main/java/org/sonar/server/component/ws/ProjectMeasuresQueryValidator.java

```diff
@@ -41,12 +42,44 @@ public ProjectMeasuresQueryValidator(DbClient dbClient) {
 
   public void validate(DbSession dbSession, ProjectMeasuresQuery query) {
     List<String> metricKeys = new ArrayList<>(query.getMetricCriteria().stream().map(MetricCriterion::getMetricKey).collect(Collectors.toSet()));
```

Comment:
not sure why you just do a `Collectors.toList()` instead of `new ArrayList<>(…)`?